### PR TITLE
[Fix #599] Add Python documentation

### LINKED ISSUE #599: Add Python support documentation
Sonata workflow supports embedded execution of python scripts. 
This functionality ought to be documented as part of existing custom function types and under integration directory

### PR DESCRIPTION
**Pull request by fjtirado**
Fixes https://github.com/apache/incubator-kie-kogito-docs/issues/599

**Description:**
Add python documentation

### CODE REVIEW

File: serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/integrations/custom-functions-python.adoc [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    }
+  ]
+----
+
+Once done, you can use that function several times to execute arbitrary python code. The python code is provided as argument of the function call through `script` property. 
+
+[source,json]
+----
+"functionRef": 
+   "name" : "Imports",
```
Comment (gmunozfe) [inline serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/integrations/custom-functions-python.adoc:43]: Shouldn't it be lowercase following python function naming convention?

**Review by gmunozfe — APPROVED** [review]
Just cosmetic comments, good job @fjtirado !

File: serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/integrations/custom-functions-python.adoc (hunk 1 above)
Comment (fjtirado) [reply serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/integrations/custom-functions-python.adoc:43]: Actually this name should not be there at all ;)

**Review by ricardozanini — APPROVED** [review]

**Comment by ricardozanini** [comment]
@kaldesai can you please take a look?

File: serverlessworkflow/modules/ROOT/pages/core/custom-functions-support.adoc [hunk 2]
```diff
@@ -300,6 +300,10 @@ The Camel route is responsible to produce the return value in a way that the wor
 
 include::../../pages/_common-content/camel-valid-responses.adoc[]
 
+[[con-func-python]
+== Python custom function
+{product_name} provides an implementation of a custom function to execute embedded Python scripts and functions. See xref:use-cases/advanced-developer-use-cases/integrations/custom-functions-knative.adoc[Invoking Python from {product_name}]
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/core/custom-functions-support.adoc:305]:
```suggestion
{product_name} implements a custom function to execute embedded Python scripts and functions. See xref:use-cases/advanced-developer-use-cases/integrations/custom-functions-knative.adoc[Invoking Python from {product_name}]
```

File: serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/integrations/custom-functions-python.adoc [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+:compat-mode!:
+// Metadata:
+:description: Describe Python execution capabilities
+:keywords: kogito, workflow, quarkus, serverless, python, AI
+
+This document describes how to integrate python scripts and functions into your workflow using {product_name} custom functions. The code appearing in this document is copied from link:{kogito_sw_examples_url}/serverless-workflow-python-quarkus[`serverless-workflow-python-quarkus`] example application and link:{kogito_runtimes_url}/quarkus/addons/python/integration-tests/src/main/resources/PythonService.sw.json[PythonService] integration test.
+
+== Enable Python support
+
+ To enable Python support, you need to the python add-on dependency to your {prouct_name} module `pom.xml` file
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/integrations/custom-functions-python.adoc:11]:
```suggestion
 To enable Python support, you must add the Python add-on dependency to your {prouct_name} module `pom.xml` file
```

File: serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/integrations/custom-functions-python.adoc [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+----
+<dependency>
+      <groupId>org.apache.kie.sonataflow</groupId>
+      <artifactId>sonataflow-addons-quarkus-python</artifactId>
+</dependency>
+----
+
+== Invoking embedded Python script.
+
+{product_name} supports of execution python script in the same memory address than the running workflow.
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/integrations/custom-functions-python.adoc:23]:
```suggestion
{product_name} supports the execution of Python script in the same memory address as the running workflow.
```

File: serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/integrations/custom-functions-python.adoc [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+
+To invoke a python script the first step is to define a custom python function at the beginning of the flow.
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/integrations/custom-functions-python.adoc:25]:
```suggestion
To invoke a Python script the first step is to define a custom Python function at the beginning of the flow. 
```

File: serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/integrations/custom-functions-python.adoc [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+ "functions": [
+    {
+      "name": "python",
+      "type": "custom",
+      "operation": "script:python"
+    }
+  ]
+----
+
+Once done, you can use that function several times to execute arbitrary python code. The python code is provided as argument of the function call through `script` property.
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/integrations/custom-functions-python.adoc:38]:
```suggestion
Once done, you can use that function several times to execute arbitrary Python code. The Python code is provided as an argument of the function call through the `script` property. 
```

File: serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/integrations/custom-functions-python.adoc [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+----
+"functionRef": 
+    "refName": "python",
+    "arguments": {
+      "script": "import numpy as np"
+     }
+ }
+----
+
+Previous snippet imports link:https://numpy.org/[numpy] library. The same python function can be invoked again to generate an array containing three random numbers between 0 and 10.
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/integrations/custom-functions-python.adoc:50]:
```suggestion
Previous snippet imports link:https://numpy.org/[numpy] library. The same Python function can be invoked again to generate an array containing three random numbers between `0` and `10`. 
```

File: serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/integrations/custom-functions-python.adoc [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+----
+"functionRef": {
+   "refName": "python",
+   "arguments": {
+      "script": "rng = np.random.default_rng().integers(low=0,high=10,size=3)"
+    }
+  }
+----
+
+To access the result of the embedded python invocation, {product_name} provides an special context variable: `$WORKFLOW.python`. Therefore, if you want to set `rng` variable from previous script as `output` property of the workflow model, you write
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/integrations/custom-functions-python.adoc:62]:
```suggestion
To access the result of the embedded python invocation, {product_name} provides a special context variable: `$WORKFLOW.python`. Therefore, if you want to set the `rng` variable from the previous script as the `output` property of the workflow model, you write 
```

File: serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/integrations/custom-functions-python.adoc [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+"stateDataFilter" : {
+  "output" : "{result:$WORKFLOW.python.rng}"
+}
+----
+
+== Invoking Python function. 
+
+You can also invoke functions from standard or custom python modules. 
+
+You need to define a serverless workflow function definition that invokes the python function. You should specific, within `operation` property, the name of the python module and function to be invoked when the function is called. You should separate the module name and the function name using `::` and prefix them with `services::python:`
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/integrations/custom-functions-python.adoc:75]:
```suggestion
You must define a serverless workflow function definition that invokes the Python function. You should specify, within the `operation` property, the name of the Python module and function to be invoked when the function is called. You should separate the module name and the function name using `::` and prefix them with `services::python:`
```

File: serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/integrations/custom-functions-python.adoc [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+
+The following example defines a function that invokes standard python function link:https://www.geeksforgeeks.org/python-math-factorial-function/[math.factorial(x)]
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/integrations/custom-functions-python.adoc:77]:
```suggestion
The following example defines a function that invokes a standard Python function link:https://www.geeksforgeeks.org/python-math-factorial-function/[math.factorial(x)]
```

File: serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/integrations/custom-functions-python.adoc [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+
+[source,json]
+----
+ "functionRef" : {
+          "refName": "factorial",
+          "arguments" : ".x"
+       }
+----
+
+The return value of the function can be handled as any other function result using `actionDataFilter.toStateData` Serverless Workflow construct. The following will set a workflow model property called `result` with the factorial invocation returned value.
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/integrations/custom-functions-python.adoc:97]:
```suggestion
The return value of the function can be handled as any other function result using the `actionDataFilter.toStateData` Serverless Workflow construct. The following will set a workflow model property called `result` with the factorial invocation returned value. 
```

File: serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/integrations/custom-functions-python.adoc [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+[source,json]
+----
+  "actionDataFilter" : {
+        "toStateData" : ".result"
+      }
+----
+
+== Further reading 
+
+The link:{kogito_sw_examples_url}/serverless-workflow-openvino-quarkus[Openvino] illustrates the powerful AI capabilities of integrating workflows with Python. It is a must seen for all interested on the topic.
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/integrations/custom-functions-python.adoc:108]:
```suggestion
The link:{kogito_sw_examples_url}/serverless-workflow-openvino-quarkus[Openvino] illustrates the powerful AI capabilities of integrating workflows with Python. It is a must-see for all interested in the topic. 
```

**Review by kaldesai — APPROVED** [review]
Minor suggestions! Otherwise, your suggestions look good to me.

**Pull request merged**